pandas/tests/groupby/test_missing.py

```diff
@@ -43,6 +43,17 @@ def test_ffill_missing_arguments():
         df.groupby("b").fillna()
 
 
+def test_ffill_with_string_column():
+    # GH 40250
+    result = (
+        DataFrame({"a": pd.array([None, "a"], dtype="string"), "b": [0, 0]})
+        .groupby("b")
+        .ffill()
+    )
+    expected = DataFrame({"a": pd.array([None, "a"], dtype="string")})
+    tm.assert_frame_equal(result, expected)
+
+
```

Comment:
we *may* have a similar example with object dtype, if you can co-locate, alternatively can parameterize on the dtype as well.

Reply:
after #41343 is merged, can use the `any_string_dtype` fixture